### PR TITLE
For #26606 - Add backplating to Top Sites

### LINKED ISSUE #26606: Add back plating to Top Sites
Add backplating UI according to the [Figma](https://www.figma.com/file/d7l1iGabQYqnLcnQbHDGzM/Wallpaper-2.0-%2B-Onboarding-v106?node-id=5536%3A189429). Existing functionality should remain.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21366)

### PR DESCRIPTION
**Pull request by MozillaNoah**
| Before | After |
|--|--|
|  |  |
|  |  |

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #26606

### CODE REVIEW

**Review by gabrielluong — CHANGES_REQUESTED** [review]
^

**Comment by gabrielluong** [comment]
Let's also ask for a UX review prior to the next round of code review.

**Comment by MozillaNoah** [comment]
@AndiAJ, I saw your PR to add/update the tests for Top Sites. I needed to update the UI of the Top Sites items with this PR, so would it be possible to get some help fixing-up `assertSponsoredTopSiteIsDisplayed` after your changes in https://github.com/mozilla-mobile/fenix/pull/26639 land? (I believe that's the function where the test is failing here)

**Comment by AndiAJ** [comment]
Hi @MozillaNoah , thanks for the heads-up! 😄 
In this case I'll put #26639 on hold until the UI for the top sites items are updated.

If `verifySponsoredShortcutsListTest` is blocking you, I would say to disable it (please open an " 💔 Intermittent UI Test Issue" for it and add the disabled label so we have it on our radar)
 
I'll try to fix it in my PR after your changes land.

I'm on PTO and will be back only next Tuesday, if you encounter any issues @AaronMT or @sv-ohorvath can help you out.

**Comment by MozillaNoah** [comment]
> Hi @MozillaNoah , thanks for the heads-up! 😄 In this case I'll put #26639 on hold until the UI for the top sites items are updated.
> 
> If `verifySponsoredShortcutsListTest` is blocking you, I would say to disable it (please open an " 💔 Intermittent UI Test Issue" for it and add the disabled label so we have it on our radar)
> 
> I'll try to fix it in my PR after your changes land.
> 
> I'm on PTO and will be back only next Tuesday, if you encounter any issues @AaronMT or @sv-ohorvath can help you out.

@AndiAJ OK, I'll disable it then. This won't land until Monday anyway. Thank you!!

**Comment by MozillaNoah** [comment]
Recent push disables a failing UI test as a result of the design updates (opened https://github.com/mozilla-mobile/fenix/issues/26698) and centers the Top Sites within their allotted horizontal space.

**Comment by MozillaNoah** [comment]
Recent push fixes the clipped shadows at the top of each top site. This was most noticeable from the sites in the top row.

The screenshots in the root comment have also been updated.

**Comment by Kate-Galetski** [comment]
It looks great with 4dp.

File: app/src/main/res/values/styles.xml
Comment (gabrielluong) [inline app/src/main/res/values/styles.xml:674]: Can you point me to where these values are coming from? Usually we would want a top sites design to present a second row to know what these values should be

**Review by gabrielluong — APPROVED** [review]
This is approval for the code changes. I don't think it requires a re-review if we had to adjust any values. However, I do want us to check in UX about the 11sp font size. Another concern I do have is the fact that the amount text displayed is a bit less given the width of the item being smaller. For example in the screenshots you provided, before you could see all of "Home Depot" and "Top Articles", but you see much less with the smaller card width. So, I want to surface this as a UX review (that may involve more than 1 UX) to see if that is a concern.

File: app/src/main/res/values/styles.xml
Comment (MozillaNoah) [reply app/src/main/res/values/styles.xml:674]: margin top was introduced as a piece of feedback from Kate, since the top shadow was cut off. Adding a margin of to the top of the card to pad-out the item in grid was the way to fully show the shadow. 4 was the agreed upon value as per [this comment above](https://github.com/mozilla-mobile/fenix/pull/26614#issuecomment-1233189900) .

Margin bottom is the original value from Top Sites's original implementation in 2020, and seems to still be correct [per the figma](https://www.figma.com/file/d7l1iGabQYqnLcnQbHDGzM/Wallpaper-2.0-%2B-Onboarding-v106?node-id=5536%3A189429).

Comment (gabrielluong) [reply app/src/main/res/values/styles.xml:674]: 👍

File: app/src/main/res/layout/top_site_item.xml
Comment (MozillaNoah) [reply app/src/main/res/layout/top_site_item.xml:64]: Agreed. I'll double check this value.

Comment (MozillaNoah) [reply app/src/main/res/layout/top_site_item.xml:64]: to close this loop, 11 is the correct value

**Comment by MozillaNoah** [comment]
I'll land this on Monday

**Pull request merged**
